explainability-core/src/main/java/org/kie/trustyai/metrics/drift/fouriermmd/FourierMMD.java

```diff
@@ -0,0 +1,504 @@
+package org.kie.trustyai.metrics.drift.fouriermmd;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+
+import org.apache.commons.math3.distribution.NormalDistribution;
+import org.apache.commons.math3.linear.Array2DRowRealMatrix;
+import org.apache.commons.math3.random.JDKRandomGenerator;
+import org.apache.commons.math3.random.RandomGenerator;
+import org.apache.commons.math3.stat.descriptive.moment.Mean;
+import org.apache.commons.math3.stat.descriptive.moment.StandardDeviation;
+import org.kie.trustyai.explainability.model.Dataframe;
+import org.kie.trustyai.explainability.model.Value;
+
+public class FourierMMD {
+
+    private boolean deltaStat;
+    private int n_test;
+    private int n_window;
+    private double sig;
+    private int randomSeed;
+    private int n_mode;
+    private double threshold;
+    private double gamma;
+
+    private Mean mean = new Mean();
+
+    private boolean isBiasCorrected = false;
+    private StandardDeviation std = new StandardDeviation(isBiasCorrected);
+
+    private NormalDistribution normalDistribution = new NormalDistribution();
+
+    private FourierMMDFitting fitStats = new FourierMMDFitting();
+
+    public FourierMMD(
+            double threshold,
+            int n_window,
+            int n_test,
+            int n_mode,
+            int randomSeed,
+            double sig,
+            boolean deltaStat,
+            double gamma) {
+        this.deltaStat = deltaStat;
+        this.n_test = n_test;
+        this.n_window = n_window;
+        this.sig = sig;
+        this.n_mode = n_mode;
+        this.threshold = threshold;
+        this.randomSeed = randomSeed;
+        this.gamma = gamma;
+    }
+
+    public FourierMMD(Dataframe dfTrain) {
+
+        this(0.8, 168, 100, 512, 1234, 10.0, true, 1.5);
+
+        precompute(dfTrain);
+    }
+
+    public FourierMMD(FourierMMDFitting fourierMMDFitting) {
+
+        this(0.8, 168, 100, 512, 1234, 10.0, true, 1.5);
+
+        fitStats = fourierMMDFitting;
+    }
+
+    // def learn(self, data: pd.DataFrame):
+    public FourierMMDFitting precompute(Dataframe data) {
```

Comment:
This is a _long_ function, any way to break it up?

Reply:
I factored a bunch of code out of it.  If you want I can break it up more....